test/suites/scale/provisioning_test.go

```diff
@@ -135,6 +191,54 @@ var _ = Describe("Provisioning", Label(debug.NoWatch), Label(debug.NoEvents), fu
 		},
 		)
 
+		env.MeasureProvisioningDurationFor(func() {
+			By("waiting for the deployment to deploy all of its pods")
+			env.ExpectCreated(deployment)
+			env.EventuallyExpectPendingPodCount(selector, replicas)
+
+			By("kicking off provisioning by applying the nodePool and nodeClass")
+			env.ExpectCreated(nodePool, nodeClass)
+
+			env.EventuallyExpectCreatedNodeClaimCount("==", expectedNodeCount)
+			env.EventuallyExpectCreatedNodeCount("==", expectedNodeCount)
+			env.EventuallyExpectInitializedNodeCount("==", expectedNodeCount)
+			env.EventuallyExpectHealthyPodCount(selector, replicas)
+		}, map[string]string{
+			aws.TestCategoryDimension:           testGroup,
+			aws.TestNameDimension:               "pod-dense",
+			aws.ProvisionedNodeCountDimension:   strconv.Itoa(expectedNodeCount),
+			aws.DeprovisionedNodeCountDimension: strconv.Itoa(0),
+			aws.PodDensityDimension:             strconv.Itoa(replicasPerNode),
+		})
+	}, SpecTimeout(time.Minute*30))
+	It("should scale successfully on a pod-dense scale-up with minValues in the NodePool requirement", func(_ context.Context) {
+		replicasPerNode := 110
+		maxPodDensity := replicasPerNode + dsCount
+		expectedNodeCount := 60
+		replicas := replicasPerNode * expectedNodeCount
+		deployment.Spec.Replicas = lo.ToPtr[int32](int32(replicas))
+		nodePool.Spec.Template.Spec.Kubelet = &corev1beta1.KubeletConfiguration{
+			MaxPods: lo.ToPtr[int32](int32(maxPodDensity)),
+		}
+		test.ReplaceRequirements(nodePool, corev1beta1.NodeSelectorRequirementWithMinValues{
+			// With Prefix Delegation enabled, .large instances can have 434 pods.
+			NodeSelectorRequirement: v1.NodeSelectorRequirement{
+				Key:      v1beta1.LabelInstanceSize,
+				Operator: v1.NodeSelectorOpIn,
+				Values:   []string{"large"},
+			},
+		},
+		)
+
+		test.ReplaceRequirements(nodePool, corev1beta1.NodeSelectorRequirementWithMinValues{
+			// minValues is restricted to 30 to have enough instance types to be sent to launch API and not make this test flaky.
+			NodeSelectorRequirement: v1.NodeSelectorRequirement{
+				Key:      v1.LabelInstanceTypeStable,
+				Operator: v1.NodeSelectorOpExists,
+			},
+			MinValues: lo.ToPtr(30),
+		})
+
```

Comment:
Can we combine these two functions? `test.ReplaceRequirements` is a variadic function

Reply:
Done.